# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/ipc/DocumentChannelParent.cpp
# Commit: 3998aeb0935c
# Full Hash: 3998aeb0935c84872b9eb63e74cc5563f8736f6b
# Author: Jean-Yves Avenard <jyavenard@mozilla.com>
# Date: 2020-06-12 14:56:55
# Regressor Bug: 1641737
# File Overlap Count: 1
# Description:
#   Bug 1641737 - P5. Remove unnecessary method. r=mattwoodrow
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D78210
# ==============================================================================

diff -r f6c685e4b841 -r 3998aeb0935c netwerk/ipc/DocumentChannelParent.cpp
--- a/netwerk/ipc/DocumentChannelParent.cpp	Fri Jun 12 05:56:39 2020 +0000
+++ b/netwerk/ipc/DocumentChannelParent.cpp	Fri Jun 12 05:56:46 2020 +0000
@@ -79,8 +79,7 @@
         self->mDocumentLoadListener = nullptr;
       },
       [self](DocumentLoadListener::OpenPromiseFailedType&& aRejectValue) {
-        if (aRejectValue.mStatus == NS_ERROR_DOCSHELL_DYING) {
-          Unused << self->SendDeleteSelf();
+        if (!self->CanSend()) {
           return;
         }
         Unused << self->SendDisconnectChildListeners(